controllers/managedocs_controller.go

```diff
@@ -428,8 +428,17 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 			// Deleting OCS CSV from the namespace
 			r.Log.Info("deleting OCS CSV")
 			if csv, err := r.getCSVByPrefix(ocsOperatorName); err == nil {
-				if err := r.delete(csv); err != nil {
-					return ctrl.Result{}, fmt.Errorf("unable to delete csv: %v", err)
+				// There's a possibility that csv will be deleted in previous reconcile and
+				// getCSVByPrefix will return emtpy csv in those cases, so need to check for value
+				// ex: setting a finalizer in next step might fail due to resourceVersion changes
+				// IOW, there might be resource modification contention, if we aren't checking for
+				// csv value then err will be returned everytime leading to continued reconcile loops
+				if csv != nil {
```

Comment:
I agree with the comment above, but I think the fix needs to go inside `getCSVByPrefix`. if at the end of the loop no CSV was found then we should return a NotFound error to match the behavior of the k8s client `get` operation

Reply:
ack, then I'll need to check against the error right?

Reply:
We are already doing it in the original code

Reply:
addressed in https://github.com/red-hat-storage/ocs-osd-deployer/pull/165